jetty-ee10/jetty-ee10-maven-plugin/src/main/java/org/eclipse/jetty/ee10/maven/plugin/MavenWebAppContext.java

```diff
@@ -282,15 +283,21 @@ public void doStart() throws Exception
 
         // Set up the classes dirs that comprises the equivalent of
         // WEB-INF/classes
-        if (_testClasses != null)
+        if (_testClasses != null && _testClasses.exists())
             _webInfClasses.add(_testClasses);
-        if (_classes != null)
+        if (_classes != null && _classes.exists())
             _webInfClasses.add(_classes);
 
         // Set up the classpath
-        _classpathFiles = new ArrayList<>();
-        _classpathFiles.addAll(_webInfClasses);
-        _classpathFiles.addAll(_webInfJars);
+        _classpathUris = new ArrayList<>();
+        _webInfClasses.forEach(f -> _classpathUris.add(f.toURI()));
+        _webInfJars.forEach(f ->
+        {
+            // ensure our JAR file references are `jar:file:...` URI references
+            URI jarFileUri = Resource.toJarFileUri(f.toURI());
```

Comment:
Since `Resource.toJarFileUri` is completely `Resource`-agnostic, it can probably be moved to `URIUtil`.

Reply:
Ditto, the future FileID/URIUtil PR.